simconvoi.cc

```diff
@@ -1895,12 +1895,16 @@ void convoi_t::ziel_erreicht()
 				if(  halt.is_bound() &&  gr->get_weg_ribi(v->get_waytype())!=0  ) {
 					halt->book(1, HALT_CONVOIS_ARRIVED);
 				}
+				// the direction of the waiting vehicle is same? opposite?
+				// reference direction to detect leading or following
+				ribi_t::ribi ribi_reference_direction = (  ( v->get_convoi()->get_next_initial_direction()  &  v->get_convoi()->front()->get_direction() ) > 0  ) ? ribi_reference_direction = v->get_direction(): ribi_reference_direction = ribi_t::backward(v->get_direction());
+				bool coupling_is_leading = !( ( self->front()->get_direction() & ribi_reference_direction ) > 0 );
```

Comment:
- こういうことでしょうか？
```suggestion
				bool coupling_is_leading = (self->front()->get_direction() & ribi_reference_direction )==0;
```

- `const` をつけてください。
- 変数名ですが、この変数は「連結相手が先頭」という意味にはならないと思います。自編成先頭の向きと連結相手のnext_initial_directionが異なる->自分が先頭にならないといけないという状態なので、 `should_this_convoy_be_parent` でしょうか？

Reply:
変数名を修正し、対応しました